modules/build/src/main/scala/scala/build/Artifacts.scala

```diff
@@ -83,6 +85,25 @@ object Artifacts {
     logger: Logger
   ): Either[BuildException, Artifacts] = either {
 
+    addClang.map { _ =>
+      val cache = FileCache()
+      val task = {
+
+        val artifact = Artifact(url = CLangInstaller.mambaBinaryUrl
+        // checksumUrls TODO: verify check sum of archive
+        ).withChanging(true)
+        cache.file(artifact).run.flatMap {
+          case Left(e: Exception) => Task.fail(e)
```

Comment:
NIT Just in case, so that we keep this line in the stack traces we would get. We also don't need the `: Exception` annotation (I think it's actually an `ArtifactError`, but better not put type annotations like this in matches, when the match doesn't need to do an `asInstanceOf` under-the-hood)
```suggestion
          case Left(e) => Task.fail(new Exception(e))
```